tests/unit/misc/test_crashdialog.py

```diff
@@ -77,3 +79,25 @@ def test_invalid_text(self):
         text = INVALID_CRASH_TEXT.strip().replace('_', ' ')
         typ, func = crashdialog.parse_fatal_stacktrace(text)
         assert (typ, func) == ('', '')
+
+
+@pytest.mark.parametrize('env, expected', [
+    ({'FOO': 'bar'}, ""),
+    ({'FOO': 'bar', 'LC_ALL': 'baz'}, "LC_ALL = baz"),
+    ({'LC_ALL': 'baz', 'PYTHONFOO': 'fish'}, "LC_ALL = baz\nPYTHONFOO = fish"),
+    (
+        {'DE': 'KDE', 'DESKTOP_SESSION': 'plasma'},
+        "DE = KDE\nDESKTOP_SESSION = plasma"
+    ),
+    (
+        {'QT5_IM_MODULE': 'fcitx', 'QT_IM_MODULE': 'fcitx'},
+        "QT_IM_MODULE = fcitx"
+    ),
+    ({'LANGUAGE': 'foo', 'LANG': 'en_US.UTF-8'}, "LANG = en_US.UTF-8"),
+])
+def test_get_environment_vars(monkeypatch, env, expected):
+    """Test for crashdialog._get_environment_vars."""
+    os.environ = {}
```

Comment:
I don't think this will work - you're just changing where `os.environ` points to, but not actually deleting the items from the "existing" `os.environ` mapping.

You can probably do something like this:

```python
for key in os.environ:
    monkeypatch.delenv(key)
```